src/dask_awkward/layers/layers.py

```diff
@@ -225,14 +237,20 @@ def prepare_for_projection(self) -> tuple[AwkwardInputLayer, TypeTracerReport, T
             The black-box state object returned by the IO function.
         """
         assert self.is_projectable
+        fn = maybe_unwrap(self.io_func)
         new_meta_array, report, state = cast(
-            ImplementsProjection, self.io_func
+            ImplementsProjection, fn
         ).prepare_for_projection()
 
+        if io_func_rer_wrapped(self.io_func):
+            new_return = (new_meta_array, type(new_meta_array)([]))
```

Comment:
Is this just creating a blank `ak.Array`?
```suggestion
            new_return = (new_meta_array, ak.from_iter([]))
```

Reply:
Yes it is, for historical reasons awkward wasn't imported in this module, but I think it would be fine now.